ci/update-version

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+usage() { cat << EOF
+Usage: $0 [-hcv]
+
+Update version stored in pyproject.toml file. Optionally, commit change.
+New version must be greater than the previous version.
+
+Options:
+    -h  Display this information.
+    -c  Specify the change should be committed. (Optional)
+    -v  A specific version to be used in updating nextflow.config and to apply as a tag if requested.
+EOF
+}
+
+COMMIT=""; VERSION=""
+while getopts "hcv:" OPTION; do
+    case "$OPTION" in
+        h) usage; exit 1;;
+        c) COMMIT=true;;
+        v) VERSION=${OPTARG};;
+        \?) usage; (>&2 echo "Exception: Unknown option $OPTION."); exit 1;;
+    esac
+done
+
+if [ -z "${VERSION}" ];
+then
+    usage; (>&2 echo "Exception: No version supplied."); exit 1
+fi
+
+if [ -z "$(git tag --merged main --sort=taggerdate)" ];
+then
+    echo "No previous version found, proceeding with ${VERSION}"
+else
+    PREVIOUS_VERSION=$(git tag --merged main --sort=taggerdate | grep "^[0-9\.]\+$" | tail -n 1)
+    if ! printf "%s\n%s\n" "${PREVIOUS_VERSION}" "${VERSION}" | sort -V -C; then
+        echo "ERROR ${VERSION} is less than the previous version, ${PREVIOUS_VERSION}." 1>&2
+        exit 1
+    fi
+fi
+
+echo "Updating pyproject.toml to version: ${VERSION}"
+
+case "$OSTYPE" in
+    darwin*|bsd*)
+        echo "Using BSD sed style"
+        sed_no_backup=( -i '' )
+        ;;
+    *)
+        echo "Using GNU sed style"
+        sed_no_backup=( -i )
+        ;;
+esac
+
+sed -E "${sed_no_backup[@]}" "s/^(version[[:space:]]+=[[:space:]])\"(.*)\"/\1\"${VERSION}\"/" pyproject.toml
```

Comment:
I realize if poetry is available you could do: `poetry version ${NEW_VERSION}`. And it will Just Work!

Reply:
Poetry is way cleaner! That's a great solution.